### ISSUE TITLE
Add OCI based helm repositories support

### ISSUE THREAD

**Issue by loktionovam** [crumbhole/argocd-lovely-plugin#89]
Hello.

I have a dependency chart in Chart.yaml

```yaml
dependencies:
- name: service
  version: 0.1.0
  repository: oci://gitlab.example.com/repo/main/chart
```

and get the error `Error: looks like "oci://gitlab.example.com/repo/main/chart" is not a valid chart repository or cannot be reached: object required`

It looks like the argocd-lovely-plugin doesn't support OCI based helm registries https://helm.sh/docs/topics/registries/ because here https://github.com/crumbhole/argocd-lovely-plugin/blob/main/pkg/processor/helmProcessor.go#L42 the plugin uses `helm repo add --force-update` command which doesn't support OCI registries (https://github.com/helm/helm/issues/10565)

Can you add support of OCI based helm repositories to your plugin?

**Comment by Joibel** [comment]
Thanks for the report - haven't used OCI repos myself so haven't hit it. This doesn't seem too hard. I'll see what I can do.

**Comment by Joibel** [comment]
I have made a change to attempt to help with this, but I don't know of any public OCI repo to test against. @loktionovam , if you know of one, please let me know, otherwise the next release will have this change in.

**Comment by RichardWardle** [comment]
Hey @Joibel 

You should be able to test with (what i have done below) - https://gallery.ecr.aws/karpenter/karpenter

```
.
├── README.MD
├── helm
│   ├── Chart.yaml
│   ├── env.txt
│   └── values.yaml
└── kustomize
    ├── kustomization.yml
    ├── nodetemplate.yml
    └── provisioners.yml
```

Chart.yaml looks like:
```
apiVersion: v2
version: 1.0.0
name: karpenter
dependencies:
- name: karpenter
  version: v0.22.1
  repository: oci://public.ecr.aws/karpenter
```

When using stable we also see the same issue as loktionovam:

```
rpc error: code = Unknown desc = `argocd-lovely-plugin` failed exit status 1: 2023/01/14 01:47:39 exit status 1: Error: looks like "oci://public.ecr.aws/karpenter" is not a valid chart repository or cannot be reached: object required
```

When using your latest push ghcr.io/crumbhole/argocd-lovely-plugin:5c28271c2630001784a23828adc34c5261dc6784 I see the following (hard refresh done and app deleted and recreated to avoid any caching issues):

```
rpc error: code = Unknown desc = `argocd-lovely-plugin` failed exit status 1: 2023/01/14 01:58:04 exit status 1: Error: no repositories found. You must add one before updating
```

If you remove the oci:// from the Chart.yaml it complains which is to be expected: Error: could not find protocol handler for:

Deploying the helm chart straight from ArgoCD works fine (currently we deploy Karpenter via helm and then use another app to deploy our config using wave sync precedence). A Snip of our argocd app looks similar to:

```
  source:
    chart: karpenter
    repoURL: public.ecr.aws/karpenter
    targetRevision: v0.22.1
    helm:
      values: |
        serviceAccount:
```

Happy to provide any other information you may want or test another version!

**Comment by loktionovam** [comment]
@Joibel  Hello, you can use any OCI docker registry, for example dockerhub 

```
helm push  ./cloudnative-pg-0.16.0.tgz oci://registry-1.docker.io/loktionovam                                                                                                                               
Pushed: registry-1.docker.io/loktionovam/cloudnative-pg:0.16.0
Digest: sha256:86aaa0adb010b805ef0dea49cdeaa6eabe9a235b16c7d90d91f81e726be9b1fb

helm pull oci://registry-1.docker.io/loktionovam/cloudnative-pg                                                                                                                                             
Pulled: registry-1.docker.io/loktionovam/cloudnative-pg:0.16.0
Digest: sha256:86aaa0adb010b805ef0dea49cdeaa6eabe9a235b16c7d90d91f81e726be9b1fb
```

**Comment by Joibel** [comment]
Thanks @loktionovam and @RichardWardle, I think it's fixed now, I've released a 0.13.0 with the change in. Please let me know either way, and re-open this if it isn't working for you.

**Comment by RichardWardle** [comment]
> Thanks @loktionovam and @RichardWardle, I think it's fixed now, I've released a 0.13.0 with the change in. Please let me know either way, and re-open this if it isn't working for you.

Tested this against the karpenter OCI repo and worked well - thanks heaps @Joibel!
